images/tests/eks/e2e_test.go

```diff
@@ -0,0 +1,92 @@
+//go:build e2e
+// +build e2e
+
+package test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	ts "github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+func TestAws(t *testing.T) {
+	t.Parallel()
+
+	setupPath := "./setup/"
+	defer ts.RunTestStage(t, "teardown", func() {
+		terraform.Destroy(t, ts.LoadTerraformOptions(t, setupPath))
+	})
+	ts.RunTestStage(t, "setup", func() {
+		opts := &terraform.Options{TerraformDir: setupPath}
+		ts.SaveTerraformOptions(t, setupPath, opts)
+		terraform.InitAndApply(t, opts)
+	})
+	setupOpts := ts.LoadTerraformOptions(t, setupPath)
+
+	// Group tests that require an EKS cluster to exists
+	t.Run("eks", func(t *testing.T) {
+		t.Run("aws-load-balancer-controller", func(t *testing.T) {
+			t.Parallel()
+
+			opts := &terraform.Options{
+				TerraformDir: "../../aws-load-balancer-controller/e2e",
+				Vars: map[string]interface{}{
+					"name":                      terraform.Output(t, setupOpts, "name"),
+					"cluster_name":              terraform.Output(t, setupOpts, "cluster_name"),
+					"cluster_oidc_provider_arn": terraform.Output(t, setupOpts, "cluster_oidc_provider_arn"),
+					"public_subnets":            terraform.OutputList(t, setupOpts, "public_subnets"),
+					"private_subnets":           terraform.OutputList(t, setupOpts, "private_subnets"),
+					// TODO: This can be replaced with an actual image build once we plumb outputs through
+					"image_ref": "cgr.dev/chainguard/aws-load-balancer-controller:latest",
+				},
+			}
+
+			down, err := tUp(t, opts)
+			if err != nil {
+				t.Fatalf("Failed to create ALB: %s", err)
+			}
+			defer down()
+
+			t.Run("should have a valid ALB", func(t *testing.T) {
```

Comment:
I like that we can get as complicated as we want here, either with go, or with more terraform orchestrated through terratest